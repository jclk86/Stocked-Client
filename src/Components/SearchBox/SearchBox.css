@import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600&display=swap");
.search_inventory_filter {
  text-align: center;
  max-width: 30%;
  height: 25px;
  font-size: 16px;
  font-weight: bold;
  font-family: "Open sans", sans-serif;
}

.container_search_filter {
  margin-top: 50px;
  max-width: 100%;
  text-align: center;
}

@media only screen and (max-width: 600px) {
  .search_inventory_filter {
    max-width: 50%;
  }
}
