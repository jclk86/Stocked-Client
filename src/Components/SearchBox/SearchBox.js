import React, { Component } from "react";
import "./SearchBox.css";

export default class SearchBox extends Component {
  static defaultProps = {
    updateSearch: () => {}
  };

  render() {
    return (
      <div className="container_search_filter">
        <label htmlFor="search_filter"></label>
        <input
          id="search_filter"
          name="search_filter"
          onChange={e => this.props.updateSearch(e.target.value)}
          type="text"
          placeholder="search"
          className="search_inventory_filter"
        ></input>
      </div>
    );
  }
}
