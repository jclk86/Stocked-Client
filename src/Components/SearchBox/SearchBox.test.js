import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import sinon from "sinon";
import SearchBox from "./SearchBox";

describe(`SearchBox Component`, () => {
  const handleChangeSpy = sinon.spy();
  it("Renders SearchBox without issue", () => {
    const wrapper = shallow(<SearchBox />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("filters inventory list based on user input", () => {
    const wrapper = mount(<SearchBox onChange={handleChangeSpy}></SearchBox>);
    const event = { target: { name: "search_filter", value: "Apple" } };
    wrapper.find("input").simulate("change", event);
  });
});
