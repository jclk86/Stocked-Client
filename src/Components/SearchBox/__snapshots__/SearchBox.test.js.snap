// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component Renders SearchBox without issue 1`] = `
<div
  className="container_search_filter"
>
  <label
    htmlFor="search_filter"
  />
  <input
    className="search_inventory_filter"
    id="search_filter"
    name="search_filter"
    onChange={[Function]}
    placeholder="search"
    type="text"
  />
</div>
`;
